docs/openfaas-pro/jetstream.md

```diff
@@ -182,34 +183,60 @@ Important configuration flags:
 
   - The queue-worker automatically extends the ack window for functions that require more time to complete. In order to prevent us from having to extend the ack window to often we recommend configuring a default acknowledgement waiting time of 3 minutes. This can be configured with the `--wait` flag.
 
-  - The `--max-waiting` flag limits the number of subscribers a queue-worker can create. The value should be at least `max_inflight * queue-worker-replicas`.
-
-    If you are running 3 replicas of the queue-worker with a max_inflight setting of 300 this value should be 900.
-
-  - The `--max-pending` flag limits the number of messages that can have a pending status. Messages that are queued for retries are also considered pending. The value of this flag should be at least `max_inflight * queue-worker-replicas + buffer`. The size of the buffer depends on the number of retries your queue needs to be able to handle. Set this to `-1` to allow any number of pending messages.
+  - The `--max-pending` flag limits the number of messages that can have a pending status. Messages that are queued for retries are also considered pending. It is set to `-1` to allow an unlimited number of pending messages by default. The consumer is paused and no new messages are delivered when this limit is reached. If you select a customer value it should be at least `max_inflight * queue-worker-replicas + buffer`. The size of the buffer depends on the number of retries your queue needs to be able to handle.
 
     This value can always be update later:
 
     ```bash
     nats consumer edit --max-pending 6000
     ```
+  
+  - The `--max-waiting` flag limits the number of subscribers a queue-worker can create. We recommend using the default value of 512 but the value should be at least equal to the number of queue worker replicas.
 
 #### Configure the queue-worker
 As a final step the queue-worker needs to be configured to use the externally created Stream and Consumer.
 
-For the shared OpenFaaS queue edit the values.yaml file of the OpennFaaS chart. The name of the Consumer used by the queue-worker is set with `jetstreamQueueWorker.durableName`. The name of the Stream needs to be set with `nats.channel`.
+For the shared OpenFaaS queue edit the values.yaml file of the OpenFaaS chart. The name of the Consumer used by the queue-worker is set with `jetstreamQueueWorker.durableName`. The name of the Stream needs to be set with `nats.channel`.
 
 ```yaml
 jetstreamQueueWorker:
   durableName: faas-workers
 
 nats:
   streamReplication: 1
-  channel: faas-request
+  channel: faas-request-workers
 ```
 
 A dedicated queue using the [queue-worker Helm chart](https://github.com/openfaas/faas-netes/tree/master/chart/queue-worker) can be configured by setting the `nats.stream.name` and `nats.consumer.durableName` parameters.
 
+## Upgrade the stream retention policy
```

Comment:
This is a one-off migration step, so would it make sense to make it a customer community announcement instead?

Reply:
Upgrade the stream retention policy => Reset the stream for async messages.

Then we can remove references to i.e. LimitsPolicy etc and make it generic.

> From time to time, you may wish to reset or purge the stream for async messages. Either due to a change in the policy, or configuration of the stream, or because you have generated a large number of unnecessary messages, etc.

Reply:
There is a separate PR for this: https://github.com/openfaas/docs/pull/368